fabric-item-api-v1/src/main/java/net/fabricmc/fabric/api/item/v1/ShieldRegistry.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.item.v1;
+
+import net.minecraft.item.ItemConvertible;
+
+import net.fabricmc.fabric.impl.item.ShieldRegistryImpl;
+
+/**
+ * Registry for defining an item as a shield.
+ * Shields should also override several {@code use} methods to work properly (see {@link net.minecraft.item.ShieldItem} for reference).
+ */
+public final class ShieldRegistry {
+	private ShieldRegistry() { }
+
+	/**
+	 * @param item the item to define as shield
+	 */
+	public static void add(ItemConvertible item) {
+		add(item, -1);
+	}
+
+	/**
+	 * @param item the item to define as shield
+	 * @param axeDisableDuration how long the cooldown lasts when the shield gets hit by an axe (0 to disable, -1 for vanilla duration)
+	 */
+	public static void add(ItemConvertible item, int axeDisableDuration) {
+		ShieldRegistryImpl.add(item, new Entry(axeDisableDuration));
+	}
+
+	/**
+	 * @param item the item to get from the registry
+	 * @return the {@link Entry} from the shield registry or null if the item is not registered
+	 */
+	public static Entry get(ItemConvertible item) {
+		return ShieldRegistryImpl.get(item);
+	}
+
+	public static final class Entry {
```

Comment:
I feel somewhat mixed about a publicly exposed entry class. Getters for each value tracked the the shield and getting from impl is probably safer.

Reply:
So would you move the Entry class to impl or ditch it altogether and go with the Object2IntMap?

Reply:
He is more like moving entry to impl so user never know there is an entry class. but Imo having an entry is good as it can reduce extra map lookups. Hence I disagree with @i509VCB.

Reply:
Should clarify I meant make the entry class an impl detail and just have getters that refer to entry class object in impl.

Reply:
> just have getters that refer to entry class object in impl.

Still, calling these methods separately will impose extra map lookups. Also should there be different attributes to a shield, they tend to associated closely (i.e. polled together than separately as I see). say if we add custom attribute of shield's damage block chance, that value is more likely polled together with the custom cooldown time as they are more likely used together.

Hence, I support @florensie's current setup where he just exposes the entry. exposing the constructor now I think is fine as well (or even just make the entry mutable) as it would allow us to update attributes of an item as a shield

Reply:
Also map lookup? We already store the object mixed into item.

If we add any future shield options like the bonk sound, the current entry class is effectively immutable. So we would need to make it mutable if public, which is going to over expose module impl details.